src/TriggerBinding/SqlTriggerUtils.cs

```diff
@@ -111,5 +113,17 @@ internal static async Task<int> GetUserTableIdAsync(SqlConnection connection, Sq
                 return (int)userTableId;
             }
         }
+
+        /// <summary>
+        /// Returns the formatted leases table name. If userDefinedLeasesTableName is null, the default name Leases_{FunctionId}_{TableId} is used.
+        /// </summary>
+        /// <param name="userDefinedLeasesTableName">Leases table name defined by the user</param>
+        /// <param name="userTableId">SQL object ID of the user table</param>
+        /// <param name="userFunctionId">Unique identifier for the user function</param>
+        internal static string GetLeasesTableName(string userDefinedLeasesTableName, string userFunctionId, int userTableId)
+        {
+            return string.IsNullOrEmpty(userDefinedLeasesTableName) ? string.Format(CultureInfo.InvariantCulture, LeasesTableNameFormat, $"{userFunctionId}_{userTableId}") :
+                string.Format(CultureInfo.InvariantCulture, UserDefinedLeasesTableNameFormat, $"{userDefinedLeasesTableName}");
```

Comment:
This still needs to be quote-escaped

Reply:
Do you mean to call userDefinedLeasesTableName.AsSingleQuoteEscapedString() here?

Reply:
AsBrackQuotedString since we're quoting it with brackets here, but yes (and remove the [] from the format string since it will be added by that call)